packages/react/src/component-tests/IcButton/IcButton.cy.tsx

```diff
@@ -0,0 +1,663 @@
+/* eslint-disable react/jsx-no-bind */
+/// <reference types="Cypress" />
+
+import React from "react";
+import { IcButton } from "../../components";
+import { CYPRESS_AXE_OPTIONS } from "../../../cypress/utils/a11y";
+import { mount } from "cypress/react";
+import {
+  DARK_BG_STYLE,
+  SlottedIcon,
+  SlottedIconNoViewBox,
+  SlottedRightIcon,
+} from "./IcButtonTestData";
+import { NOT_BE_CALLED_ONCE, HAVE_VALUE } from "../utils/constants";
+
+const DEFAULT_TEST_THRESHOLD = 0.03;
+
+const WIN_CONSOLE_SPY = "@spyWinConsoleLog";
+
+const ID_NAME = "#name";
+const TEXT_INPUT = "@textInput";
+
+const submitForm = (ev: React.FormEvent<HTMLFormElement>) => {
+  ev.preventDefault();
+  console.log("foo");
+};
+
+describe("IcButton", () => {
+  it("renders", () => {
+    mount(<IcButton>Test</IcButton>);
+
+    cy.get("ic-button").contains("Test").should("be.visible");
+  });
+
+  it("should pass onclick method", () => {
+    mount(
+      <IcButton
+        onClick={(e) => {
+          e.currentTarget.innerText = "clicked";
+        }}
+      >
+        Test
+      </IcButton>
+    );
+    cy.checkHydrated("ic-button");
+    cy.clickOnButton("ic-button");
```

Comment:
Presume this needs some sort of assert here to check it's been a success?
`cy.get("#ic-button").should("have.text", "clicked");`